crawl-ref/source/map-knowledge.cc

```diff
@@ -326,3 +326,31 @@ bool map_cell::update_cloud_state()
     // TODO: track decay & vanish appropriately (based on some worst case?)
     return false;
 }
+
+FixedVector<coord_def, 2> known_map_bounds() {
+    int min_x = GXM, max_x = 0, min_y = 0, max_y = 0;
```

Comment:
Do you know why they handle x and y differently here? It seems to me like they could be perfectly symmetric.

Reply:
It's because the outer loop is over y coordinates, so the y's we encounter are monotonically increasing. That means the first hit has to be the minimum and the last hit has to be the maximum.